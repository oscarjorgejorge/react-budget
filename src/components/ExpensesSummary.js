import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import expenseTotal from "../selectors/expenses-total";
import selectExpenses from "../selectors/expenses";
import numeral from "numeral";

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  // console.log(expenseCount);
  // console.log(expensesTotal);

  const expenseWord = expenseCount === 1 ? "expense" : "expenses";
  const formattedExpensesTotal = numeral(expensesTotal / 100).format("$0,0.00");
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">
          Viewing <span>{expenseCount}</span> {`${expenseWord} `}
          totalling <span> {formattedExpensesTotal}</span>
          <div className="page-header__actions">
            <Link className="button" to="/create">
              Add Expense
            </Link>
          </div>
        </h1>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  console.log(visibleExpenses);

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: expenseTotal(visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
