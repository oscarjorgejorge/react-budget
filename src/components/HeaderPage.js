import React from "react";
import { NavLink } from "react-router-dom";
import { connect } from "react-redux";
import { startLogout } from "../actions/auth";

export const Header = ({ startLogout }) => (
  <header>
    <h1>Expensify</h1>
    <button onClick={startLogout}>Log Out</button>
    <div>
      <NavLink to="/" activeClassName="is-active" exact={true}>
        Go home
      </NavLink>
    </div>
    <div>
      <NavLink to="/create" activeClassName="is-active">
        Create
      </NavLink>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout()),
});

export default connect(undefined, mapDispatchToProps)(Header);
