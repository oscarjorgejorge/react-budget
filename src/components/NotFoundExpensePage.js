import React from "react";
import { Link } from "react-router-dom";

const NotFoundExpensePage = () => (
  <div>
    404! - <Link to="/">Go home</Link>
  </div>
);

export default NotFoundExpensePage;
