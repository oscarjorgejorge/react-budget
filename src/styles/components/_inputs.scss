.text-input {
    border: 1px solid #cacccd;
    height: 50px;
    font-size: $font-size-large;
    font-weight: 300;
    padding: $s-size;
}

.select {
    @extend .text-input;
    padding: 0.5rem;
}

.textarea {
    @extend .text-input;
    height: 10rem;
}