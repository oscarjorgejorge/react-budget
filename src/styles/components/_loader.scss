.loader {
    align-items: center;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw;
}

.loader_image {
    height: 6rem;
    width: 6rem;
}