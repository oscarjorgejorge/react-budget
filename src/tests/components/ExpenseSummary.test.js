import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("should render ExpensesSummary with 1 expense", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={235} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should render ExpensesSummary with multiple expenses", () => {
  const wrapper = shallow(<ExpensesSummary count={3} total={456178} />);
  expect(wrapper).toMatchSnapshot();
});
