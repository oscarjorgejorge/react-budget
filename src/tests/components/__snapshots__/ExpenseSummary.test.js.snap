// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
    totalling 
    $2.35
  </p>
</div>
`;

exports[`should render ExpensesSummary with multiple expenses 1`] = `
<div>
  <p>
    Viewing 
     
    expenses
    totalling 
    $0.00
  </p>
</div>
`;
